src/main/java/uk/gov/pay/ledger/queue/sqs/SqsQueueService.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.pay.ledger.queue.sqs;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.AmazonSQSException;
+import com.amazonaws.services.sqs.model.ChangeMessageVisibilityRequest;
+import com.amazonaws.services.sqs.model.DeleteMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.ledger.app.LedgerConfig;
+import uk.gov.pay.ledger.queue.QueueException;
+import uk.gov.pay.ledger.queue.QueueMessage;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class SqsQueueService {
+    private final Logger logger = LoggerFactory.getLogger(SqsQueueService.class);
+
+    private AmazonSQS sqsClient;
+
+    private final int messageMaximumWaitTimeInSeconds;
+    private final int messageMaximumBatchSize;
+
+    @Inject
+    public SqsQueueService(AmazonSQS sqsClient, LedgerConfig ledgerConfig) {
+        this.sqsClient = sqsClient;
+        this.messageMaximumBatchSize = ledgerConfig.getSqsConfig().getMessageMaximumBatchSize();
+        this.messageMaximumWaitTimeInSeconds = ledgerConfig.getSqsConfig().getMessageMaximumWaitTimeInSeconds();
+    }
+
+    public List<QueueMessage> receiveMessages(String queueUrl, String messageAttributeName) throws QueueException {
+        try {
+            ReceiveMessageRequest receiveMessageRequest = new ReceiveMessageRequest(queueUrl);
+            receiveMessageRequest
+                    .withMessageAttributeNames(messageAttributeName)
+                    .withWaitTimeSeconds(messageMaximumWaitTimeInSeconds)
+                    .withMaxNumberOfMessages(messageMaximumBatchSize);
+
+            ReceiveMessageResult receiveMessageResult = sqsClient.receiveMessage(receiveMessageRequest);
+
+            return QueueMessage.of(receiveMessageResult);
+        } catch (AmazonSQSException | UnsupportedOperationException e) {
+            logger.error("Failed to receive messages from SQS queue - [{}] {}", e.getClass().getCanonicalName(), e.getMessage());
+            throw new QueueException(e.getMessage());
```

Comment:
This would propagate the original exception as a "cause", which means that the original stack trace is preserved:

```suggestion
            throw new QueueException("Failed to receive messages from SQS queue", e);
```

Reply:
Makes sense 👍 
(I was even thinking about it - no idea why I haven't done it 😅 )